Source/Particles/WarpXParticleContainer.cpp

```diff
@@ -642,18 +706,192 @@ WarpXParticleContainer::DepositCharge (WarpXParIter& pti, RealVector& wp,
+        WARPX_PROFILE_VAR_START(blp_sort);
+        amrex::DenseBins<ParticleType> bins;
+        {
+            const Geometry& geom = Geom(lev);
+            const auto dxi = geom.InvCellSizeArray();
+            const auto plo = geom.ProbLoArray();
+            const auto domain = geom.Domain();
+
+            auto& ptile = ParticlesAt(lev, pti);
+            auto& aos   = ptile.GetArrayOfStructs();
+            auto pstruct_ptr = aos().dataPtr();
+
+            ParticleTileType ptile_tmp;
+            ptile_tmp.define(ptile.NumRuntimeRealComps(), ptile.NumRuntimeIntComps());
+            ptile_tmp.resize(ptile.numParticles());
+
+            Box box = pti.validbox();
+            box.grow(1);
+            amrex::IntVect bin_size = WarpX::sort_bin_size;
+            int ntiles = numTilesInBox(box, true, bin_size);
+
+            bins.build(ptile.numParticles(), pstruct_ptr, ntiles,
+                       [=] AMREX_GPU_HOST_DEVICE (const ParticleType& p) -> unsigned int
+                       {
+                           Box tbx;
+                           auto iv = getParticleCell(p, plo, dxi, domain);
+                           AMREX_ASSERT(box.contains(iv));
+                           auto tid = getTileIndex(iv, box, true, bin_size, tbx);
+                           return static_cast<unsigned int>(tid);
+                       });
+
+            gatherParticles(ptile_tmp, ptile, ptile.numParticles(), bins.permutationPtr());
+            ptile.swap(ptile_tmp);
+        }
+        WARPX_PROFILE_VAR_STOP(blp_sort);
+
+        // get tile boxes
+        amrex::Gpu::DeviceVector<Box> tboxes(bins.numBins(), amrex::Box());
+        {
+            const Geometry& geom = Geom(lev);
+            const auto dxi = geom.InvCellSizeArray();
+            const auto plo = geom.ProbLoArray();
+            const auto domain = geom.Domain();
+
+            auto& ptile = ParticlesAt(lev, pti);
+            auto& aos   = ptile.GetArrayOfStructs();
+            auto pstruct_ptr = aos().dataPtr();
+
+            Box box = pti.validbox();
+            box.grow(1);
+            amrex::IntVect bin_size = WarpX::sort_bin_size;
+
+            const auto offsets_ptr = bins.offsetsPtr();
+            auto tbox_ptr = tboxes.dataPtr();
+            amrex::ParallelFor(bins.numBins(),
+                               [=] AMREX_GPU_DEVICE (int ibin) {
+                                   const int bin_start = offsets_ptr[ibin];
+                                   const int bin_stop = offsets_ptr[ibin+1];
+                                   if (bin_start < bin_stop) {
+                                       const auto& p = pstruct_ptr[bin_start];
+                                       Box tbx;
+                                       auto iv = getParticleCell(p, plo, dxi, domain);
+                                       AMREX_ASSERT(box.contains(iv));
+                                       auto tid = getTileIndex(iv, box, true, bin_size, tbx);
```

Comment:
```suggestion
                                       [[maybe_unused]] auto tid = getTileIndex(iv, box, true, bin_size, tbx);
```

Reply:
I think you might have accidentally resolved instead of applied this change.

Reply:
I'm fixing it on my branch, and will push in a batch in a few minutes, to not spin CI